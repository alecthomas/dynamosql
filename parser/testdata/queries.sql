SELECT * FROM movies
SELECT title, year FROM movies
SELECT title, year FROM movies WHERE title = "The Dark Knight"
SELECT title, year FROM movies WHERE title = "The Dark Knight" AND year >= 2009
SELECT title, year FROM movies WHERE title = "The Dark Knight" AND year BETWEEN 2009 AND 2015
SELECT title, year FROM movies WHERE title = "The Dark Knight" AND year BETWEEN 2009 AND 2015 AND actor = "Will Smith"
SELECT title, year FROM movies WHERE title = "The Dark Knight" AND (year BETWEEN 2009 AND 2015 OR actor = "Will Smith")
SELECT * FROM movies WHERE title = :title
SELECT * FROM movies WHERE title = :title AND attribute_exists(year)
SELECT * FROM movies WHERE title = :title AND begins_with(actor, "Will")
