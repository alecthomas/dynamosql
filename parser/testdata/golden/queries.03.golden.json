{
  "Query": "SELECT title, year FROM movies WHERE title = \"The Dark Knight\" AND year \u003e= 2009",
  "AST": {
    "Expression": {
      "All": false,
      "Projections": [
        "title",
        "year"
      ]
    },
    "From": {
      "Table": "movies",
      "Where": {
        "And": [
          {
            "Parenthesized": null,
            "Not": null,
            "Operand": {
              "Operand": {
                "Value": {
                  "PlaceHolder": null,
                  "Number": null,
                  "String": null,
                  "Boolean": null,
                  "Null": false
                },
                "SymbolRef": {
                  "Symbol": "title"
                }
              },
              "ConditionRHS": {
                "Compare": {
                  "Operator": "=",
                  "Operand": {
                    "Value": {
                      "PlaceHolder": null,
                      "Number": null,
                      "String": "The Dark Knight",
                      "Boolean": null,
                      "Null": false
                    },
                    "SymbolRef": null
                  }
                },
                "Between": null,
                "In": null
              }
            },
            "Function": null
          },
          {
            "Parenthesized": null,
            "Not": null,
            "Operand": {
              "Operand": {
                "Value": {
                  "PlaceHolder": null,
                  "Number": null,
                  "String": null,
                  "Boolean": null,
                  "Null": false
                },
                "SymbolRef": {
                  "Symbol": "year"
                }
              },
              "ConditionRHS": {
                "Compare": {
                  "Operator": "\u003e=",
                  "Operand": {
                    "Value": {
                      "PlaceHolder": null,
                      "Number": 2009,
                      "String": null,
                      "Boolean": null,
                      "Null": false
                    },
                    "SymbolRef": null
                  }
                },
                "Between": null,
                "In": null
              }
            },
            "Function": null
          }
        ]
      }
    },
    "Limit": null
  }
}